### PR TITLE
Update docker guide with Jib content

### PR DESCRIPTION
**Pull request by barchetta**
Incorporates content from #265

### CODE REVIEW

**Comment by barchetta** [comment]
@cricket007 FYI

File: docs/src/main/docs/guides/05_Dockerfile.adoc [hunk 1]
```diff
@@ -281,13 +287,143 @@ modules to the `jlink` command:
...
+
+https://github.com/GoogleContainerTools/jib[Jib] is a collection of build tools for
+constructing Docker images for Java applications. Jib allows you to build images
+straight from a Maven (or Gradle) plugin without the need for the `docker`
+command. It also uses a
+https://github.com/GoogleContainerTools/distroless[distroless] base image that
+can result in a smaller final image.
+
+We'll be using the `jib-maven-plugin` in this example. With
+`jib-maven-plugin` you don't need a Dockerfile
```
Comment (cricket007) [inline docs/src/main/docs/guides/05_Dockerfile.adoc]: nit: `[,] you don't need`

File: docs/src/main/docs/guides/05_Dockerfile.adoc [hunk 2]
```diff
@@ -281,13 +287,143 @@ modules to the `jlink` command:
...
+and it creates a final Docker image that follows a set
+of best-practices for packaging Java applications in a container.
+
+Docker is only required if you are going to install the image locally.
```
Comment (cricket007) [inline docs/src/main/docs/guides/05_Dockerfile.adoc]: Maybe clarify ... 

1. `docker` (the command) is only required... (put code ticks)

2. The Docker _application_ is only required...

File: docs/src/main/docs/guides/05_Dockerfile.adoc [hunk 3]
```diff
@@ -281,13 +287,143 @@ modules to the `jlink` command:
...
+It is *not* required to build and _push the image_ to a Docker Registry,
+ such as Docker Hub. This can be accomplished via `mvn jib:build` after
+https://github.com/GoogleContainerTools/jib/tree/master/jib-maven-plugin#configuration[configuring which
+Registry to connect to].
+This fits well in an environment where administrative permissions
+ are unavailable on developer machines to install or run Docker.
+
+Add the following plugin to the `pom.xml`. This adds a goal to
```
Comment (cricket007) [inline docs/src/main/docs/guides/05_Dockerfile.adoc]: nit: Transitioning: `[To add Jib to this example,] add the following plugin`

File: docs/src/main/docs/guides/05_Dockerfile.adoc [hunk 4]
```diff
@@ -281,13 +287,143 @@ modules to the `jlink` command:
...
+                      <jmxFlag>-Djava.awt.headless=true</jmxFlag>
+                      <jmxFlag>-XX:+UnlockExperimentalVMOptions</jmxFlag>
+                      <jmxFlag>-XX:+UseCGroupMemoryLimitForHeap</jmxFlag>
+                      <jmxFlag>-XX:InitialRAMFraction=2</jmxFlag>
+                      <jmxFlag>-XX:MinRAMFraction=2</jmxFlag>
+                      <jmxFlag>-XX:MaxRAMFraction=2</jmxFlag>
+                      <jmxFlag>-XX:+UseG1GC</jmxFlag>
+                      <jmxFlag>-XX:MaxGCPauseMillis=100</jmxFlag>
+                      <jmxFlag>-XX:+UseStringDeduplication</jmxFlag>
+                  </jvmFlags>
```
Comment (cricket007) [inline docs/src/main/docs/guides/05_Dockerfile.adoc]: These JVM flags could probably be removed, and left as an excersize for the reader.

File: docs/src/main/docs/guides/05_Dockerfile.adoc (hunk 1 above)
Comment (barchetta) [reply docs/src/main/docs/guides/05_Dockerfile.adoc]: Fixed

File: docs/src/main/docs/guides/05_Dockerfile.adoc (hunk 2 above)
Comment (barchetta) [reply docs/src/main/docs/guides/05_Dockerfile.adoc]: Fixed.

File: docs/src/main/docs/guides/05_Dockerfile.adoc (hunk 3 above)
Comment (barchetta) [reply docs/src/main/docs/guides/05_Dockerfile.adoc]: Fixed.

File: docs/src/main/docs/guides/05_Dockerfile.adoc (hunk 4 above)
Comment (barchetta) [reply docs/src/main/docs/guides/05_Dockerfile.adoc]: Done.

**Comment by cricket007** [comment]
Actually, should probably keep

```xml
<jmxFlag>-XX:+UseG1GC</jmxFlag>
```

**Pull request merged**